packages/nimble-components/src/select/index.ts

```diff
@@ -510,13 +520,21 @@ export class Select
         }
 
         if (!this.options?.includes(focusTarget as ListboxOption)) {
+            let currentActiveIndex = this.openActiveIndex ?? this.selectedIndex;
             this.open = false;
-            if (this.selectedIndex === -1) {
+            if (currentActiveIndex === -1) {
                 this.selectedIndex = this.indexWhenOpened!;
+                currentActiveIndex = this.selectedIndex;
+                if (this.selectedIndex >= 0) {
+                    (
+                        this.options[this.selectedIndex]! as ListOption
+                    ).activeOption = true;
+                }
             }
 
-            if (this.indexWhenOpened !== this.selectedIndex) {
+            if (this.indexWhenOpened !== currentActiveIndex) {
                 this.updateValue(true);
+                this.selectedIndex = currentActiveIndex;
```

Comment:
Is there ever a time that `this.selectedIndex` shouldn't be updated when `this.updateValue()` is called? Should the setting of `this.selectedIndex` be moved into `this.updateValue()` to ensure they always stay in sync?

Reply:
So, the idea is that `updateValue` should be called in response to `selectedIndex` changing, as its implementation is optimized to pull the value to set from the current `selectedIndex`. If there is a refactor to be done to provide a more confident synchronization between these two fields, I'd rather hold that for another time.